diff --git a/runtime/lib/ThreadData.cpp b/runtime/lib/ThreadData.cpp
index 4e11668..18fa4d5 100644
--- a/runtime/lib/ThreadData.cpp
+++ b/runtime/lib/ThreadData.cpp
@@ -42,8 +42,10 @@ namespace WinPixEventRuntime
 
     PIXEventsThreadInfo* ThreadData::GetPixEventsThreadInfo()
     {
+#if DBG
         // This is our thread info that's only meant to be used by this thread.
         assert(m_threadId == std::this_thread::get_id());        
+#endif
 
         if (m_isEnabled)
         {
@@ -85,8 +87,10 @@ namespace WinPixEventRuntime
 
     uint64_t ThreadData::ReplaceBlock(std::optional<uint64_t> const& eventTime)
     {
+#if DBG
         // This is our thread info that's only meant to be used by this thread.
         assert(m_threadId == std::this_thread::get_id());
+#endif
 
         if (auto oldBlock = Flush(eventTime))
         {
diff --git a/runtime/lib/WinPixEventRuntime.cpp b/runtime/lib/WinPixEventRuntime.cpp
index d8d6bb8..463bd23 100644
--- a/runtime/lib/WinPixEventRuntime.cpp
+++ b/runtime/lib/WinPixEventRuntime.cpp
@@ -158,7 +158,7 @@ namespace WinPixEventRuntime
 //
 
 
-UINT64 PIXEventsReplaceBlock(PIXEventsThreadInfo* threadInfo, bool getEarliestTime) noexcept
+UINT64 WINAPI PIXEventsReplaceBlock(PIXEventsThreadInfo* threadInfo, bool getEarliestTime) noexcept
 {
     std::optional<uint64_t> eventTime;
     if (getEarliestTime)
